rfd/0058-package-distribution.md

```diff
@@ -0,0 +1,66 @@
+---
+authors: Fred Heinecke (fred.heinecke@goteleport.com)
+state: draft
+---
+
+# RFD 58 - Package Distribution
+
+## What
+
+Hosting and configuration of APT and YUM repositories.
+
+## Why
+
+Currently we are building .deb and .rpm file in Teleport's Drone pipeline via reprepro and createrepo, then publishing to AWS S3. Between the tools used and the pipeline's current configuration there are two main issues: a lack of channel support for major versions, and an inability to host multiple minor versions for each major version. By fixing these issues we can allow customers to upgrade the teleport package to the latest minor version release for their major version, as well as allow them to roll back to a previous minor version if need be.
+
+## Details
+
+Several potential solutions were investigated and their features compared as shown below:
+
+| Feature/Product                            | Current solution           | Current solution with fixes                                                                                  | JFrog Artifactory                                                       | PackageCloud                                   | GCP Artifact Registry                                                             |
```

Comment:
"Current solution" and "Current solution with fixes" should be clarified. 
Even as an insider, I don't know what they mean. I might instead say "reprepro + s3 + cloudfront" (current).  And then clarify what the fixes are.

Reply:
Yeah, I'm still unclear about what engineering effort is required for the fixes, and where would it go.

I mean, "fixes" can mean anything from fixing a typo in a promotion script to writing & and hosting a custom distribution service from scratch, and the tradeoffs for each are vastly different.

Reply:
Added some notes regarding this here: https://github.com/gravitational/teleport/blob/rfd/0058-package-distribution/rfd/0058-package-distribution.md#research.